types/src/transaction/pricing_mode.rs

```diff
@@ -102,90 +103,38 @@ impl Display for PricingMode {
 }
 
 impl ToBytes for PricingMode {
-    fn write_bytes(&self, writer: &mut Vec<u8>) -> Result<(), bytesrepr::Error> {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
         match self {
             PricingMode::Classic {
                 payment_amount,
                 gas_price_tolerance: gas_price,
                 standard_payment,
-            } => {
-                CLASSIC_TAG.write_bytes(writer)?;
-                payment_amount.write_bytes(writer)?;
-                gas_price.write_bytes(writer)?;
-                standard_payment.write_bytes(writer)
-            }
-            PricingMode::Reserved { receipt } => {
-                RESERVED_TAG.write_bytes(writer)?;
-                receipt.write_bytes(writer)
-            }
+            } => serialize_classic_pricing_mode(payment_amount, gas_price, standard_payment),
```

Comment:
Looking at this side-by-side it does seem that enums were by default extensible by adding a new variant. What we couldn't do was add optional fields to the existing variants. Not sure it's worth the change for enums but maybe it's not obvious due to nesting.

Reply:
Well, that is an architectural decision :)